routine-data.h

```diff
@@ -2,26 +2,27 @@
 #define __ROUTINE_DATA__
 
 using getData = int();
+using Filter = int();
 
```

Comment:
we do not need Filter callback
letus do following
1. keep last 10 (DATA_SIZE) samples in memory queue
2. if new sample comes push back it and pop front 1-st of the saved measurements. this is how data queue is up to date. do this in gatherData function.
3. calculate values with ALL of the filters on the updated data and publish each into its mqtt topic. letus have "filter/median", "filter/mean" and "filter/exp_smooth" topics for filtered values and "data" for raw ,easurement (the one we've just pushed).

this is how I would desire this functionality to be refactored
and this is why we do not need filter callback - just call all of them

Reply:
I guess. that we will need filter callback here.

Without callback - we will need to publish filtered data in processData function.
Or - we can return tuple\map of filtered values from this function(This approach is not good. cause it assumes using of another container from c++ standard library).

And the best approach(in my opinion) is to use callback.
We will pass callbacks to processData function. Fetch results and then publish data in runDataRoutine function.